src/controller/proxy/controller.go

```diff
@@ -124,7 +122,24 @@ func (c *controller) ProxyManifest(ctx context.Context, p *models.Project, art l
 	}
 	// Push manifest in background
 	go func() {
-		c.waitAndPushManifest(ctx, remoteRepo, man, art, ct, r)
+		// When pull with tag, if same artifact with same tag and digest is already exist in local,
+		// log the audit log and skip to push manifest to local
+		if len(art.Tag) == 0 {
+			c.waitAndPushManifest(ctx, remoteRepo, man, art, ct, remote)
+			return
+		}
+		a, err := c.local.GetManifest(ctx, art)
+		if err != nil {
+			log.Errorf("failed to check SameArtifact, error %v", err)
+			return
+		}
+		if a == nil {
```

Comment:
what if the digest doesn't equal to the local one?

Reply:
if the digest doesn't match the local, push to local